src/Microsoft.OpenApi/Models/OpenApiExtensibleDictionary.cs

```diff
@@ -38,15 +38,31 @@ public abstract class OpenApiExtensibleDictionary<T> : Dictionary<string, T>,
         /// </summary>
         public IDictionary<string, IOpenApiExtension> Extensions { get; set; } = new Dictionary<string, IOpenApiExtension>();
 
+
+        /// <summary>
+        /// Serialize to Open Api v3.1
+        /// </summary>
+        /// <param name="writer"></param>
+        public void SerializeAsV31(IOpenApiWriter writer)
+        {
+            Serialize(writer);
+        }
+
         /// <summary>
         /// Serialize to Open Api v3.0
         /// </summary>
-        public void SerializeAsV3(IOpenApiWriter writer, OpenApiSpecVersion version = OpenApiSpecVersion.OpenApi3_0)
+        /// <param name="writer"></param>
+        public void SerializeAsV3(IOpenApiWriter writer)
         {
-            if (writer == null)
-            {
-                throw Error.ArgumentNull(nameof(writer));
-            }
+            Serialize(writer);
+        }
+        
+        /// <summary>
+        /// Serialize to Open Api v3.0
+        /// </summary>
+        public void Serialize(IOpenApiWriter writer)
+        {
+            writer = writer ?? throw Error.ArgumentNull(nameof(writer));
```

Comment:
move the body of this method to a serializeinternal method that accepts the version as a parameter so you can pass the correct version to the extensions below

Reply:
I can just update the Serialize method to accept the version as a parameter

Reply:
so the IOpenApiSerializable interface would go back to having a single Serialize method that takes a mandatory version parameter (so we don't miss passing it along at places) and implements can either have a single method with if/else when their serialization logic is simple or break apart in multiple specialized private methods?

Reply:
No. We retain the serialize methods for v2, v3 and v31. However, since most of the logic between v3 and v31 is similar, I put that common logic in a Serialize method([as per Darrel's comment](https://github.com/microsoft/OpenAPI.NET/pull/1138#discussion_r1097984754)) but it will have a mandatory version parameter.

Reply:
right, and that serialize method will be private? In that case I think we were talking about the same thing with different names in my original comment on this thread.

Reply:
Yes it will be private.